src/server/server_family.cc

```diff
@@ -3126,16 +3144,24 @@ string ServerFamily::FormatInfoMetrics(const Metrics& m, std::string_view sectio
   };
 
   auto add_repl_info = [&] {
+    std::shared_ptr<Replica> replica;
+    std::vector<std::shared_ptr<Replica>> cluster_replicas;
     bool is_master = true;
     // Thread local var is_master is updated under mutex replicaof_mu_ together with replica_,
     // ensuring eventual consistency of is_master. When determining if the server is a replica and
     // accessing the replica_ object, we must lock replicaof_mu_. Using is_master alone is
     // insufficient in this scenario.
     // Please note that we we do not use Metrics object here.
-    {
+    if (use_replica_of_v2_) {
+      // Deep copy because tl_replica might be overwritten inbetween
```

Comment:
I find confusing the "Deep copy" term here. 
1. You copy shared_ptr which is not deep copy, imho.
2. But then you `cluster_replicas` which is indeed a (deep) copy of a vector.